session/common.go

```diff
@@ -403,10 +379,66 @@ func (x commonData) Issuer() user.ID {
 // IssuerPublicKeyBytes returns binary-encoded public key of the session issuer.
 //
 // IssuerPublicKeyBytes MUST NOT be called before ReadFromV2 or Sign methods.
-func (x *commonData) IssuerPublicKeyBytes() []byte {
-	if x.sigSet {
-		return x.sig.GetKey()
+// Deprecated: use [Container.Signature] instead.
+func (x Container) IssuerPublicKeyBytes() []byte {
+	if sig := x.Signature(); sig.Scheme() >= 0 {
+		return sig.PublicKeyBytes()
 	}
+	return nil
+}
 
+// IssuerPublicKeyBytes returns binary-encoded public key of the session issuer.
+//
+// IssuerPublicKeyBytes MUST NOT be called before ReadFromV2 or Sign methods.
+// Deprecated: use [Object.Signature] instead.
+func (x Object) IssuerPublicKeyBytes() []byte {
+	if sig := x.Signature(); sig.Scheme() >= 0 {
+		return sig.PublicKeyBytes()
+	}
 	return nil
 }
+
+type signedToken interface {
+	*Container | *Object
+	SignedData() []byte
+	AttachSignature(neofscrypto.Signature)
+}
+
+// Sign calculates signature of the given [Container] or [Object] session token
+// according to the NeoFS protocol and attaches the result into it. Use
+// [HasValidSignature] for verification. Use [Issue] if issuer has not been set
+// yet.
+func Sign[T signedToken](token T, signer neofscrypto.Signer) error {
```

Comment:
Public method with private interface.

Reply:
this is normal for generic constraints, but this func wont appear as decided

Reply:
drop